internal/nerdgraph/command_query.go

```diff
@@ -44,13 +52,39 @@ keys are the variables to be referenced in the GraphQL query.
 	PreRun: client.RequireClient,
 	Run: func(cmd *cobra.Command, args []string) {
 		var variablesParsed map[string]interface{}
+		var err error
+
+		// when variableFile is defined, read and parse it
+		if strings.TrimSpace(variablesFile) != "" {
+			var variableContent []byte
+			variableContent, err = ioutil.ReadFile(variablesFile)
+			if err != nil {
+				log.Fatal("unable to read variable file")
+			}
+			err = json.Unmarshal(variableContent, &variablesParsed)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
 
-		err := json.Unmarshal([]byte(variables), &variablesParsed)
+		// parse variables passed as arguments, this will overwrite any existing variables from file
+		err = json.Unmarshal([]byte(variables), &variablesParsed)
 		if err != nil {
 			log.Fatal(err)
 		}
```

Comment:
One small recommendation: rather than allowing `variables` to override `variableFile`, it might be better UX to only allow one or the other so the behavior is more explicit. Then just have an `if/else if` to unmarshal the right variable data into `variablesParsed`.

The `Args` block above could have an explicit check as well:
```
if argsCount == 1 && strings.TrimSpace(variablesFile) != "" && strings.TrimSpace(variables) != "" {
	return errors.New("command expects either variablesFile or variables, not both")
}
```

That's all a minor nit though, and your implementation works as I'd expect. So if you'd rather keep the behavior you have that's cool too.

Reply:
I was thinking about the following use case "I have a set of defaults (file), but sometimes I want to overwrite one variable when running" For example staging environments, or specific dashboards names. 

So it is on purpose, but let me check with person who opened the ticket.

Reply:
Let's keep it as is @jbeveland27 I hope it's clear from descriptions what's happening. So this is good to merge 🥳